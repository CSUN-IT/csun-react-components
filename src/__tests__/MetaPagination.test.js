import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import MetaPagination from '../../src/components/meta/Pagination';

configure({ adapter: new Adapter() });

describe('MetaPagination', () => {
  it('Should be defined', () => {
    expect(MetaPagination).toBeDefined();
  });

  it('Renders component correctly', () => {
    const startRange = {
      start: 1,
      end: 2,
    };
    const tree = shallow(<MetaPagination startRange={startRange} activeNumber={2} />);
    expect(tree).toMatchSnapshot();
  });
});
