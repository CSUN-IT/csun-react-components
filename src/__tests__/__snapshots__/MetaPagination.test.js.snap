// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaPagination Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaPagination
    activeNumber={2}
    endRange={null}
    startRange={
      Object {
        "end": 2,
        "start": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="disabled"
        >
          <span>
            «
          </span>
        </li>,
        <li
          className="active"
        >
          <span>
            1
          </span>
        </li>,
        <li>
          <a
            href="#2"
          >
            2
          </a>
        </li>,
        <li>
          <a
            href="#3"
          >
            3
          </a>
        </li>,
        <li>
          <a
            href="#4"
          >
            4
          </a>
        </li>,
        <li
          className="disabled"
        >
          <span>
            ...
          </span>
        </li>,
        <li>
          <a
            href=""
          >
            20
          </a>
        </li>,
        <li>
          <a
            href="#21"
          >
            21
          </a>
        </li>,
        <li>
          <a
            href="#22"
            rel="next"
          >
            »
          </a>
        </li>,
      ],
      "className": "pagination",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            «
          </span>,
          "className": "disabled",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "«",
          },
          "ref": null,
          "rendered": "«",
          "type": "span",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            1
          </span>,
          "className": "active",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1",
          },
          "ref": null,
          "rendered": "1",
          "type": "span",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#2"
          >
            2
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2",
            "href": "#2",
          },
          "ref": null,
          "rendered": "2",
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#3"
          >
            3
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3",
            "href": "#3",
          },
          "ref": null,
          "rendered": "3",
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#4"
          >
            4
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "4",
            "href": "#4",
          },
          "ref": null,
          "rendered": "4",
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            ...
          </span>,
          "className": "disabled",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "...",
          },
          "ref": null,
          "rendered": "...",
          "type": "span",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href=""
          >
            20
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "20",
            "href": "",
          },
          "ref": null,
          "rendered": "20",
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#21"
          >
            21
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "21",
            "href": "#21",
          },
          "ref": null,
          "rendered": "21",
          "type": "a",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="#22"
            rel="next"
          >
            »
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "»",
            "href": "#22",
            "rel": "next",
          },
          "ref": null,
          "rendered": "»",
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="disabled"
          >
            <span>
              «
            </span>
          </li>,
          <li
            className="active"
          >
            <span>
              1
            </span>
          </li>,
          <li>
            <a
              href="#2"
            >
              2
            </a>
          </li>,
          <li>
            <a
              href="#3"
            >
              3
            </a>
          </li>,
          <li>
            <a
              href="#4"
            >
              4
            </a>
          </li>,
          <li
            className="disabled"
          >
            <span>
              ...
            </span>
          </li>,
          <li>
            <a
              href=""
            >
              20
            </a>
          </li>,
          <li>
            <a
              href="#21"
            >
              21
            </a>
          </li>,
          <li>
            <a
              href="#22"
              rel="next"
            >
              »
            </a>
          </li>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              «
            </span>,
            "className": "disabled",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "«",
            },
            "ref": null,
            "rendered": "«",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              1
            </span>,
            "className": "active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "1",
            },
            "ref": null,
            "rendered": "1",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#2"
            >
              2
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2",
              "href": "#2",
            },
            "ref": null,
            "rendered": "2",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#3"
            >
              3
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3",
              "href": "#3",
            },
            "ref": null,
            "rendered": "3",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#4"
            >
              4
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "4",
              "href": "#4",
            },
            "ref": null,
            "rendered": "4",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              ...
            </span>,
            "className": "disabled",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "...",
            },
            "ref": null,
            "rendered": "...",
            "type": "span",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href=""
            >
              20
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "20",
              "href": "",
            },
            "ref": null,
            "rendered": "20",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#21"
            >
              21
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "21",
              "href": "#21",
            },
            "ref": null,
            "rendered": "21",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="#22"
              rel="next"
            >
              »
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "»",
              "href": "#22",
              "rel": "next",
            },
            "ref": null,
            "rendered": "»",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
